libr/cons/2048.c

```diff
@@ -74,50 +74,80 @@ static void twok_move(int u, int v) {
 	}
 }
 
-static void getval(char *val0, int i, int x) {
+static void getval(bool color, char *val0, int i, int x) {
+	const char * colorarray[] = {
+		Color_WHITE,
+		Color_RED,
+		Color_GREEN,
+		Color_MAGENTA,
+		Color_YELLOW,
+		Color_CYAN,
+		Color_BLUE,
+		Color_GRAY};
 	if (twok_buf[i][x]) {
-		sprintf (val0, "%4d", 1 << twok_buf[i][x]);
+		if (color) {
+			sprintf (val0, "%s%4d"Color_RESET, colorarray [twok_buf [i][x] % 8 ], 1 << twok_buf[i][x]);
```

Comment:
use snprintf and pass val0 size or so. this format can't overflow 32 chars, in fact i doubt it will need more than 16 bytes, but its about good practices to avoid future problems in case of refactoring